src/wallet/crypter.cpp

```diff
@@ -135,13 +136,31 @@ static bool DecryptKey(const CKeyingMaterial& vMasterKey, const std::vector<unsi
     return key.VerifyPubKey(vchPubKey);
 }
 
+static bool DecryptSpendingKey(const CKeyingMaterial& vMasterKey, const std::vector<unsigned char>& vchCryptedSecret, const libzcash::PaymentAddress& address, libzcash::SpendingKey& key)
+{
+    CKeyingMaterial vchSecret;
+    if(!DecryptSecret(vMasterKey, vchCryptedSecret, address.GetHash(), vchSecret))
+        return false;
+
+    if (vchSecret.size() != 32)
```

Comment:
``vchSecret`` is a serialised key, which does in fact correspond to the memory representation of the key but that's an implementation detail (otherwise we would just copy the memory rather than deserialising). So ``sizeof(uint252)`` is not the right thing. I would be inclined to move the constant from base58.cpp into [Zcash.h](https://github.com/zcash/zcash/blob/zc.v0.11.2.latest/src/zcash/Zcash.h).